src/binary-search-tree.js

```diff
@@ -5,6 +5,39 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(val) {
+    if (val < this.value && this.left === null) {
+      this.left = new BinarySearchTree(val);
+    } else if (val < this.value && this.left !== null) {
+      this.left.insert(val);
+    } else if (val > this.value && this.right === null) {
+      this.right = new BinarySearchTree(val);
+    } else if (val > this.value && this.right !== null) {
+      this.right.insert(val);
+    }
+  }
+  contains(y) {
+    if (this.value === y) {
+      return true;
+    } else if (y > this.value && this.right !== null) {
+      return this.right.contains(y);
+    } else if (y < this.value && this.left !== null) {
+      return this.left.contains(y);
+    }
+    return false;
+  }
+  depthFirstForEach(cb) {
+    function recurr(bst) {
+      cb.call(bst, bst.value);
```

Comment:
Here, I think you'll want to check the base case of your recursive function first before doing anything else, if only to decrease the chances of the base case not being reached and blowing the call stack.